TwoPage/src/androidTest/java/com/microsoft/device/display/samples/twopage/PageSwipeTest.kt

```diff
@@ -0,0 +1,185 @@
+     * then swipes backwards through all the pages again and asserts that each page is visible at the correct time
+     */
+    private fun swipeOnePageAtATime() {
+        val pageTags = listOf(R.string.page1_tag, R.string.page2_tag, R.string.page3_tag, R.string.page4_tag)
+
+        // Swipe forwards
+        for (page in 0..3) {
+            // Assert current page is visible
+            composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsDisplayed()
+
+            // Swipe to next page
+            composeTestRule.onRoot().performGesture { swipeLeft() }
+
+            // Assert current page is no longer visible (unless on the last page)
+            when (page) {
+                3 ->
+                    composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsDisplayed()
+                else ->
+                    composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsNotDisplayed()
+            }
+        }
+
+        // Swipe backwards
+        for (page in 3 downTo 0) {
+            // Assert current page is visible
+            composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsDisplayed()
+
+            // Swipe to previous page
+            composeTestRule.onRoot().performGesture { swipeRight() }
+
+            // Assert current page is no longer visible (unless on the first page)
+            when (page) {
+                0 ->
+                    composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsDisplayed()
+                else ->
+                    composeTestRule.onNodeWithTag(composeTestRule.getString(pageTags[page])).assertIsNotDisplayed()
+            }
+        }
+    }
+
+    /**
+     * Tests that the three dual-screen page pairs ([1,2], [2,3], and [3,4]) only swipe between each other when
+     * a vertical fold is present
+     */
+    @Test
+    fun app_verticalFold_pagesSwipeWithinLimits() {
+        composeTestRule.setContent {
+            val configuration = LocalConfiguration.current
+            val windowMetrics = remember(configuration) {
+                WindowMetricsCalculator.getOrCreate()
+                    .computeCurrentWindowMetrics(composeTestRule.activity)
+            }
+            val windowWidthDp = with(LocalDensity.current) {
+                windowMetrics.bounds.width().toDp().value
+            }
+            // REVISIT: look into "ProvideWindowInsets" from accompanist library so hardcoded value can be removed
+            val insetsPadding = if (windowWidthDp > configuration.smallestScreenWidthDp) 50 else 0
+            val pageWidth = (windowWidthDp.toInt() - insetsPadding) / 2
+
+            TwoPageAppTheme {
+                TwoPageAppContent(
+                    viewWidth = pageWidth,
+                    isDualScreen = true,
+                    foldSize = 0
+                )
+            }
+        }
```

Comment:
To make sure the page widths are correctly calculated on each emulator/orientation, I added an insetsPadding variable, but I think eventually using an [insets library](https://github.com/google/accompanist/tree/main/insets) would be cleaner.

I was thinking we could revisit this after we look into insets with TwoPaneLayout, but I'm also fine with removing the test for now or any other option if you think that would be better!

Reply:
It would be more accurate by bringing `insetsPadding` for sure. For consistency, I think we should add it to both test and app itself.

Reply:
Sounds good, I'll remove it for now and then we can add it back to the app & tests once we've figured out the correct way to calculate the inset padding!